api/database/mongooseIdPlugin.js

```diff
@@ -0,0 +1,7 @@
+export default function mongooseIdPlugin(schema, options) {
+  schema.options.toJSON.transform = function (doc, ret, options) {
+    delete ret._id; // Remove internal _id property before exposing it as JSON.
```

Comment:
Exactly. As you said, is a trimmed down version and for that reason I believe this implementation is pretty solid for now. In much larger systems, `_id` could be re-evaluated.